core/src/main/scala/org/http4s/ResponseCookie.scala

```diff
@@ -117,4 +120,175 @@ final case class ResponseCookie(
 
   def clearCookie: headers.`Set-Cookie` =
     headers.`Set-Cookie`(copy(content = "", expires = Some(HttpDate.Epoch), maxAge = Some(0L)))
+
+  private def withExpires(expires: HttpDate): ResponseCookie =
+    copy(expires = Some(expires))
+
+  private def withMaxAge(maxAge: Long): ResponseCookie =
+    copy(maxAge = Some(maxAge))
+
+  private def withDomain(domain: String): ResponseCookie =
+    copy(domain = Some(domain))
+
+  private def withPath(path: String): ResponseCookie =
+    copy(path = Some(path))
+
+  private def withSecure(secure: Boolean): ResponseCookie =
+    copy(secure = secure)
+
+  private def withHttpOnly(httpOnly: Boolean): ResponseCookie =
+    copy(httpOnly = httpOnly)
+
+  private def withSameSite(sameSite: SameSite): ResponseCookie =
+    copy(sameSite = sameSite)
+
+  private def addExtension(extension: String): ResponseCookie =
+    copy(extension = self.extension.fold(Option(extension))(old => Some(old + "; " + extension)))
```

Comment:
Some of these could be generally useful and public.  Not all of them are validated except by the parser, which would permit the construction of invalid values.  Would it be confusing to expose just the safe ones?